attributes/default.rb

```diff
@@ -4,19 +4,38 @@
 #
 # Copyright 2014, 2015 Bloomberg Finance L.P.
 #
+::Chef::Node.send(:include, ConsulCookbook::Helpers)
+
 default['consul']['service_name'] = 'consul'
 default['consul']['service_user'] = 'consul'
+# Only used on Linux
 default['consul']['service_group'] = 'consul'
 
 default['consul']['bag_name'] = 'secrets'
 default['consul']['bag_item'] = 'consul'
 
-default['consul']['config']['path'] = '/etc/consul.json'
-default['consul']['config']['data_dir'] = '/var/lib/consul'
-default['consul']['config']['ca_file'] = '/etc/consul/ssl/CA/ca.crt'
-default['consul']['config']['cert_file'] = '/etc/consul/ssl/certs/consul.crt'
+case node['os']
```

Comment:
This can be simplified quite a bit. There are two methods I am thinking of, the first being simply creating a temporary value for prefix directory and passing that through to all of the attributes. The second would be a heavier modification that would use interpolation later in the code. The [File class]() provides a platform-independent method for joining paths. I'll illustrate the latter (the former will require much more testing):
```ruby
prefix_path = case node['os']
              when 'linux', 'freebsd', 'mac_os_x'
                '/etc/consul'
              when 'windows'
                program_files
              end
default['consul']['config']['path'] = File.join(prefix_path, 'consul.json')
default['consul']['config']['data_dir'] = File.join(prefix_path, 'data')
default['consul']['config']['ca_file'] = File.join(prefix_path, 'CA', 'ca.crt')
```
The more complex solution would be lazily performing string interpolation. A helper method would need to do this so that we're able to correctly modify the path separators. I think the first approach is the most sensible right now as I don't immediately have a thought on how to do it elegantly.